### PR TITLE
cmake: Fix CMake build to allow for whitespace

### LINKED ISSUE #166: Cannot compile with space in directory name
With a clone in, for example `~/source/Vulkan ExtensionLayer` I get
```

[1/2] Re-running CMake...
CMake Warning at CMakeLists.txt:44 (message):
  CMAKE_BUILD_TYPE not set.  Using Debug for dependency build type


********************************************************************************
* NOTE: Adding target vvl_update_deps to run as needed for updating            *
*       dependencies.                                                          *
********************************************************************************
-- Using glslang install located at /home/liam/source/Vulkan ExtensionLayer/external/glslang/build/install
-- Configuring done
-- Generating done
-- Build files have been written to: /home/liam/source/Vulkan ExtensionLayer/build
[10/22] Linking CXX shared library layers/libVkLayer_khronos_synchronization2.so
FAILED: layers/libVkLayer_khronos_synchronization2.so 
: && /usr/lib64/ccache/c++ -fPIC -Wpointer-arith -Wno-unused-function -Wno-sign-compare  -Wl,--version-script=/home/liam/source/Vulkan ExtensionLayer/layers/libVkLayer_khronos_synchronization2.map,-Bsymbolic,--exclude-libs,ALL -shared -Wl,-soname,libVkLayer_khronos_synchronization2.so -o layers/libVkLayer_khronos_synchronization2.so layers/CMakeFiles/VkLayer_khronos_synchronization2.dir/synchronization2.cpp.o  utils/libVkExtLayer_utils.a && :
/usr/bin/ld: cannot open linker script file /home/liam/source/Vulkan: No such file or directory
collect2: error: ld returned 1 exit status
[11/22] Linking CXX shared library layers/libVkLayer_khronos_timeline_semaphore.so
FAILED: layers/libVkLayer_khronos_timeline_semaphore.so 
: && /usr/lib64/ccache/c++ -fPIC -Wpointer-arith -Wno-unused-function -Wno-sign-compare  -Wl,--version-script=/home/liam/source/Vulkan ExtensionLayer/layers/libVkLayer_khronos_timeline_semaphore.map,-Bsymbolic,--exclude-libs,ALL -shared -Wl,-soname,libVkLayer_khronos_timeline_semaphore.so -o layers/libVkLayer_khronos_timeline_semaphore.so layers/CMakeFiles/VkLayer_khronos_timeline_semaphore.dir/hash_table.cpp.o layers/CMakeFiles/VkLayer_khronos_timeline_semaphore.dir/timeline_semaphore.c.o  utils/libVkExtLayer_utils.a && :
/usr/bin/ld: cannot open linker script file /home/liam/source/Vulkan: No such file or directory
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by juan-lunarg**
closes #166

Tested locally to reproduce issue.

### CODE REVIEW

**Review by jeremyg-lunarg — APPROVED** [review]

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -78,7 +56,7 @@ macro(AddVkLayer target LAYER_COMPILE_DEFINITIONS)
                                          INSTALL_RPATH
                                          "@loader_path/")
     else(UNIX AND NOT APPLE) # i.e.: Linux
-        set_target_properties(VkLayer_${target} PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libVkLayer_${target}.map,-Bsymbolic,--exclude-libs,ALL")
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:81]: target_link_options fixes the path issues nicely

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -119,46 +97,45 @@ set(SYNCHRONIZATION2_FILES
...
 foreach(TARGET_NAME ${TARGET_NAMES})
-	set(CONFIG_DEFINES -DINPUT_FILE="${CMAKE_CURRENT_SOURCE_DIR}/json/${TARGET_NAME}.json.in" -DVK_VERSION=${vk_full_version})
-    # If this json file is not a metalayer, get the needed properties from that target
-    if(TARGET ${TARGET_NAME})
-        set(CONFIG_DEFINES
-            ${CONFIG_DEFINES}
-            -DOUTPUT_FILE="$<TARGET_FILE_DIR:${TARGET_NAME}>/${TARGET_NAME}.json"
-            -DRELATIVE_LAYER_BINARY="${RELATIVE_PATH_PREFIX}$<TARGET_FILE_NAME:${TARGET_NAME}>")
-        # If this json file is a metalayer, make the output path match core validation, and there is no layer binary file
+    set(INPUT_FILE "${CMAKE_CURRENT_SOURCE_DIR}/json/${TARGET_NAME}.json.in")
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:101]: I imported this logic from the vulkan validation layers.

File: layers/CMakeLists.txt [hunk 3]
```diff
@@ -119,46 +97,45 @@ set(SYNCHRONIZATION2_FILES
...
-# The output file needs Unix "/" separators or Windows "\" separators On top of that, Windows separators actually need to be doubled
-# because the json format uses backslash escapes
-file(TO_NATIVE_PATH "./" RELATIVE_PATH_PREFIX)
-string(REPLACE "\\" "\\\\" RELATIVE_PATH_PREFIX "${RELATIVE_PATH_PREFIX}")
-
-set(vk_full_version "${VulkanHeaders_VERSION}")
-# Run each .json.in file through the generator We need to create the generator.cmake script so that the generator can be run at
-# compile time, instead of configure time Running at compile time lets us use cmake generator expressions (TARGET_FILE_NAME and
-# TARGET_FILE_DIR, specifically)
-file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/generator.cmake" "configure_file(\"\${INPUT_FILE}\" \"\${OUTPUT_FILE}\")")
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:131]: This line of code will also break with whitespaces

**Comment by juan-lunarg** [comment]
> CI Vulkan-ExtensionLayer build # 278 failed.

This one looks like a CI failure. I've now fixed the Windows build.

**Pull request merged**